augur/refine.py

```diff
@@ -126,7 +126,7 @@ def register_parser(parent_subparsers):
     parser.add_argument('--clock-filter-iqd', type=float, help='clock-filter: remove tips that deviate more than n_iqd '
                                 'interquartile ranges from the root-to-tip vs time regression')
     parser.add_argument('--vcf-reference', type=str, help='fasta file of the sequence the VCF was mapped to')
-    parser.add_argument('--year-bounds', type=int, nargs='+', help='specify min or max & min prediction bounds for samples with XX in year')
+    parser.add_argument('--year-bounds', type=int, nargs='+', help='specify min or max & min prediction bounds for samples with XX in date')
```

Comment:
(from commit message)
> This shouldn't have any effect in practice, except increases the ability to detect when any ambiguous date (not just ambiguous years) falls out of bounds, which results in an error.

Not saying we shouldn't make this year → date change, but thinking thru the consequences. Might it result in someone who's using `--year-bounds` to start seeing errors on existing data that before was ok because their dates with ambiguity only in month or day parts are no longer considered within bounds?

Reply:
Fair point, this is more of a breaking change than bugfix/feature. I'll reword the commit message and update the changelog accordingly.